mlflow/pyfunc/__init__.py

```diff
@@ -1675,6 +1677,37 @@ def _validate_function_python_model(python_model):
                 error_code=INVALID_PARAMETER_VALUE,
             )
 
+    if isinstance(python_model, PythonModel):
+        try:
+            init_has_model_param = "model" in inspect.signature(python_model.__init__).parameters
+            init_has_model_assignment = "self.model =" in inspect.getsource(python_model.__init__)
+            if init_has_model_param or init_has_model_assignment:
```

Comment:
to make this more robust, i was thinking we could add a `sys.getsizeof()` check for each of the params and warn if any of the params are very large. i think it's still useful to keep the string checks around to educate users about best practices though